presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -351,34 +370,67 @@ private void invalidateSchemasCache()
         schemaNamesCache.invalidateAll();
     }
 
-    private void invalidateTablesCaches()
+    private void invalidateTablesCaches(SchemaTableName table)
     {
-        columnsCache.invalidateAll();
-        tableHandleCache.invalidateAll();
-        tableNamesCache.invalidateAll();
+        invalidateColumnsCache(table);
+        invalidateTableCache(table);
     }
 
-    private void invalidateColumnsCache(ConnectorSession session, SchemaTableName table)
+    private void invalidateTableCache(SchemaTableName table)
     {
-        columnsCache.invalidate(new ColumnsCacheKey(JdbcIdentity.from(session), table));
+        invalidateCache(tableNamesCache, key -> key.schemaName.equals(Optional.of(table.getSchemaName())));
+        invalidateCache(tableHandleCache, key -> key.tableName.equals(table));
+    }
+
+    private void invalidateColumnsCache(SchemaTableName table)
+    {
+        invalidateCache(columnsCache, key -> key.table.equals(table));
+    }
+
+    private Map<String, Object> getSessionProperties(ConnectorSession session)
+    {
+        return sessionPropertiesProviders.stream()
+                .flatMap(property -> getSessionProperty(session, property))
+                .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private static Stream<Map.Entry<String, Object>> getSessionProperty(ConnectorSession session, PropertyMetadata property)
```

Comment:
Just make this return Object

Reply:
I see this marked resolved, but i do not see this resolved. Did you want to reply here why not doing this?

Reply:
Not pushed yet